contracts/apps/AragonApp.sol

```diff
@@ -57,6 +64,36 @@ contract AragonApp is AppStorage, Autopetrified, VaultRecoverable, ReentrancyGua
         );
     }
 
+    /**
+    * @dev Check whether a call to the current app can be executed or not based on the kill-switch settings
+    * @return Boolean indicating whether the call could be executed or not
+    */
+    function canExecuteCall() public view returns (bool) {
+        IKernel _kernel = kernel();
+        bytes4 selector = _kernel.shouldDenyCallingContract.selector;
+        bytes memory callData = abi.encodeWithSelector(selector, appId(), address(this));
+        bool success = address(_kernel).call(callData);
+
+        // if the call to `kernel.shouldDenyCallingApp` reverts (using an old Kernel) we consider that
```

Comment:
Let's add it to the `canPerform` method 👍

Reply:
Another thing to have in mind is that we will be hiding too much information behind the  `APP_AUTH_FAILED` error message, it could be somehow misleading.

Reply:
Yeah, the thinking behind putting it inside `Kernel.hasPermission()` was to streamline the API and reduce a bit of the gas overhead, as I imagine it would be _very_ common to put the `auth` and `killswitchProtected` modifiers together for most apps.